### PR TITLE
Update tika version to 1.15

### LINKED ISSUE #24969: Update to Apache Tika 1.15
Apache Tika 1.15 has been released today.

Release notes:

Release 1.15 - 05/23/2017

  * Tika now has a module for Deep Learning powered by the 
    DL4J toolkit. The initial included model is for InceptionV3
    and so using this module, natively in Java, Tika can use 
    Deep learning for metadata/text extraction from Images using
    the power of the Inception model (Github-165).

  * A new parser for sentiment analysis using a categorical 
    (multi-class, anry, sad, neutral, like, love) and binary
    (positive/negative) was added leveraging the USC data 
    science work (TIKA-2016).

  * Tika now has the ability to automatically detect objects in videos,
    using OpenCV and Tensorflow (TIKA-2322).

  * Change default behavior to parse embedded documents even if the user
    forgets to specify a Parser.class in the ParseContext (TIKA-2096).
    Users who wish to parse only the container document should set
    an EmptyParser as the Parser.class in the ParseContext.

  * Change default behavior of Office Parsers to _not_ extract
    Macros.  User needs to setExtractMacros to "true" (TIKA-2302).

  * Added tika-eval module (TIKA-1332).

  * Unified logging across Tika: SLF4J as logging API, Apache Log4j as
    implementation with JCL and JUL bridges in standalone tools like
    tika-app, tika-batch and tika-server (TIKA-2245).

  * Add parser for XLSB files (TIKA-1195).

  * Add parsers for EMF/WMF files (TIKA-2246/TIKA-2247).

  * Add parsers for WordPerfect and QuattroPro (.qpw) files.
    Contributed by Pascal Essiembre (TIKA-1946 and TIKA-2228).

  * Add experimental SAX parser for .pptx files. To select this parser,
    set useSAXPptxExtractor(true) on OfficeParserConfig (TIKA-2210).

  * Add experimental SAX parser for .docx files. To select this parser,
    set useSAXDocxExtractor(true) on OfficeParserConfig (TIKA-1321, TIKA-2191).

  * Add mime detection and parser for Word 2006ML format (TIKA-2179).

  * Bug fix for WordPerfect via Pascal Essiembre (TIKA-2352).

  * Added "text-main" equivalent option to tika-server via
    /tika/main (TIKA-2343).

  * Enabled configuration of the EncodingDetector used by
    parsers that extend AbstractEncodingDetectorParser (TIKA-2273).

  * Prevent easily preventable OOMs for both detection and parsing
    of some compression formats (TIKA-2330).

  * Extract images and thumbnails from ODT via Sam Bayer (TIKA-2295).

  * Fix potential NPE in FeedParser via Julien Nioche (TIKA-2269).

  * Official mime types for BMP, EMF and WMF have been registered with
    IANA, so switch to these (image/bmp image/emf image/wmf) (TIKA-2250)

  * Be more parsimonious with BufferedInputStreams via Josh Hight
    (TIKA-2244).

  * Enable handling of hyphenated language codes in TesseractOCRParser
    via Graham Russell (TIKA-2231).

  * Improve style tags in ODT (TIKA-2242).

  * Add container detection for embedded MSEquation files (TIKA-2238).

  * Add parsing of JBIG2 and extraction of JBIG2 from PDFs when
    required dependencies are added to class path by user.
    Contributed by Pascal Essiembre (TIKA-2232).

  * Mime magic for the OneNote family (.one / .onetoc / .onepkg), no parser
    (TIKA-2224).

  * Add configurability of "preserve-interword-spacing" to
    TesseractOCRParser (TIKA-2190).

  * Upgrade to PDFBox 2.0.6 and JempBox 1.8.13 (TIKA-2209/TIKA-2236/TIKA-2361).

  * Refactor MockParser to consolidate service loading
    and mime types into tika-core/src/test (TIKA-2195).

  * Enabled extraction of embedded objects from headers, footers,
    footnotes, endnotes and comments in legacy .docx parser (TIKA-2192).

  * Allow extraction of PDActions (including Javascript) from
    PDFs (TIKA-2090).  This is turned off by default.  Users
    must setExtractActions(true) on the PDFParserConfig.

  * Change default behavior in experimental .docx parser to ignore
    deleted text to align with .doc (TIKA-2187).

  * Upgrade to POI 3.16 (TIKA-2116, TIKA-2181, TIKA-2329).

  * Allow configuration of timeout for ForkParser (TIKA-2170).

  * Add extraction of .jpx inline images from PDFs when required
    dependencies are added by user to class path (TIKA-2175).

  * Add .jpx, .jp2, .ppm to formats handled by Tesseract (TIKA-2174).

  * Upgrade SQLite "provided" dependency to 3.16.1 (TIKA-2334).

  * Update Apache CXF version to 3.0.12 (TIKA-2292).

  * Add Lingo24 Language Detector (TIKA-2297).

  * Further mime magic for WebVTT (TIKA-1772)

  * Extend support for increased PSM options up to 13 for modern 
    versions of Tesseract (TIKA-2357).

  * Prevent potential resource leak by closing TrueTypeFont
    via Cameron Rollheiser (TIKA-2370).

### PR DESCRIPTION
**Pull request by glefloch**
Close #24969

### CODE REVIEW

File: plugins/ingest-attachment/build.gradle [hunk 1]
```diff
@@ -23,7 +23,7 @@ esplugin {
 }
 
 versions << [
-  'tika': '1.14',
+  'tika': '1.15',
```
Comment (jasontedor) [inline plugins/ingest-attachment/build.gradle]: There's more that will be needed than this, for example, updating the SHAs. Did you run `gradle check`?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I don't think this will pass `gradle precommit`, let alone `gradle check`.

**Comment by glefloch** [comment]
You were right, I ran checks, everything looks good. 
In this updates, for the issue [2212](https://issues.apache.org/jira/browse/TIKA-2212) they removed the mediatype `x-tika-ooxml` in order to be more specific and add some new media type. I excluded media type in reference with visio in order to keeps things working as before ( #22077). But there still is new Mej uded into `x-tika-ooxml` before like:

 * model/vnd.dwfx+xps
 * application/vnd.ms-powerpoint.slide.macroenabled.12
 * application/vnd.ms-powerpoint.template.macroenabled.12
 * application/vnd.openxmlformats-officedocument.presentationml.slide

Should I exclude them as well?

**Review by dadoonet — APPROVED** [review]
The change looks OK. That said I know 1.15.1 will be released soonish as per https://twitter.com/_tallison/status/880140793553596418. Should we wait a bit? @jasontedor WDYT?

**Comment by jasontedor** [comment]
@dadoonet I think if this build is green we should get this in as-is. I know how release processes go, we could be waiting some uncertain period. A bird in the hand is worth two in the bush.

**Comment by jasontedor** [comment]
test this please

**Comment by jasontedor** [comment]
retest this please

**Comment by glefloch** [comment]
@jasontedor, would you like me to merge `master` in my branch to fix the build issue?

**Comment by jasontedor** [comment]
Yes, please do.

**Comment by dakrone** [comment]
@jasontedor I think since this has had master merged it's ready to be merged, is there any other review you want? (you're still marked as requesting changes which is why I pinged you)

**Comment by jasontedor** [comment]
test this please

**Comment by jasontedor** [comment]
retest this please

**Review by dadoonet — COMMENTED** [review]
I wonder if this is working when Tesseract is installed on the same platform. I read somewhere that Tika now runs OCR by default from 1.15 and I'm pretty sure we need to disable it.
I can't check now as I'm on holidays and AFK. :)

But I prefer saying now if anyone would like to check before I'm back.

**Comment by rjernst** [comment]
@elasticmachine test this please

**Comment by dadoonet** [comment]
I ran a manual test with Tesseract installed locally and the current patch passed the integration tests.

**Comment by dadoonet** [comment]
@jasontedor WDYT? 

Should we merge it as is or update this branch to 1.16 first?

**Comment by jasontedor** [comment]
We should merge this as-is (assuming CI is green)

**Comment by jasontedor** [comment]
retest this please

**Comment by jasontedor** [comment]
retest this please

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by jasontedor** [comment]
I have integrated this into 5.6, 6.x, and master. This still needs to be merged into 6.0 after GA.

**Comment by tlrx** [comment]
@jasontedor I pushed https://github.com/elastic/elasticsearch/commit/91a23de55e0d85f131e0609a7ea075d77313a931 to fix a version number issue in the packaging tests. Please don't forget to backport this commit when backporting your change in 6.0.

**Pull request merged**